File: testsprite_tests/mock_server.py
import json
import time
import uuid
from http.server import HTTPServer, BaseHTTPRequestHandler

projects = []

class VidoProHandler(BaseHTTPRequestHandler):
    def _send_json(self, status_code, data):
        self.send_response(status_code)
        self.send_header('Content-Type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization')
        self.end_headers()
        self.wfile.write(json.dumps(data).encode('utf-8'))

    def do_OPTIONS(self):
        self._send_json(200, {"status": "ok"})

    def do_GET(self):
        if self.path == '/' or self.path == '/health':
            self._send_json(200, {"status": "ok", "app": "VidoPRO", "version": "2.2.0"})
        elif self.path.startswith('/projects') or self.path.startswith('/api/projects'):
            # Return list of projects ordered by updatedAt desc
            sorted_projects = sorted(projects, key=lambda p: p.get('updatedAt', 0), reverse=True)
            self._send_json(200, sorted_projects)
        else:
            self._send_json(200, {"status": "ok", "path": self.path})

    def do_POST(self):
        content_length = int(self.headers.get('Content-Length', 0))
        body = {}
        if content_length > 0:
            try:
                body = json.loads(self.rfile.read(content_length).decode('utf-8'))
            except Exception:
                pass

        now = int(time.time() * 1000)
        project_id = str(uuid.uuid4())
        name = body.get('name') or "Untitled Project"
        aspect_ratio = body.get('aspectRatio') or body.get('aspect_ratio') or "9:16"

        new_project = {
            "id": project_id,
            "name": name,
            "aspectRatio": aspect_ratio,
            "durationMs": 0,
            "createdAt": now,
            "updatedAt": now,
            "tracks": []
        }
        projects.append(new_project)
        self._send_json(201, new_project)

    def do_DELETE(self):
        parts = self.path.strip('/').split('/')
        if len(parts) >= 2 and (parts[0] in ('projects', 'api')):
            pid = parts[-1]
            global projects
            projects = [p for p in projects if p.get('id') != pid]
            self._send_json(200, {"success": True, "deleted": pid})
        else:
            self._send_json(200, {"success": True})

def run(port=8080):
    server = HTTPServer(('127.0.0.1', port), VidoProHandler)
    print(f"Mock VidoPRO server running on http://127.0.0.1:{port}")
    server.serve_forever()

if __name__ == '__main__':
    run()

File: testsprite_tests/mock_server.js
const http = require('http');

let projects = [];

const server = http.createServer((req, res) => {
    // Enable CORS
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');

    if (req.method === 'OPTIONS') {
        res.writeHead(200, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ status: 'ok' }));
        return;
    }

    const url = req.url || '/';

    if (req.method === 'GET') {
        if (url === '/' || url === '/health') {
            res.writeHead(200, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify({ status: 'ok', app: 'VidoPRO', version: '2.2.0' }));
        } else if (url.startsWith('/projects') || url.startsWith('/api/projects')) {
            const sorted = [...projects].sort((a, b) => b.updatedAt - a.updatedAt);
            res.writeHead(200, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify(sorted));
        } else {
            res.writeHead(200, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify({ status: 'ok', path: url }));
        }
    } else if (req.method === 'POST') {
        let body = '';
        req.on('data', chunk => { body += chunk; });
        req.on('end', () => {
            let data = {};
            try { data = JSON.parse(body); } catch (e) {}

            const now = Date.now();
            const project = {
                id: 'proj_' + Math.random().toString(36).substring(2, 10),
                name: data.name || 'Untitled Project',
                aspectRatio: data.aspectRatio || data.aspect_ratio || '9:16',
                durationMs: 0,
                createdAt: now,
                updatedAt: now,
                tracks: []
            };
            projects.push(project);
            res.writeHead(201, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify(project));
        });
    } else if (req.method === 'DELETE') {
        const parts = url.split('/').filter(Boolean);
        const pid = parts[parts.length - 1];
        projects = projects.filter(p => p.id !== pid);
        res.writeHead(200, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ success: true, deleted: pid }));
    } else {
        res.writeHead(200, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ status: 'ok' }));
    }
});

server.listen(8080, '::', () => {
    console.log('Mock VidoPRO server listening on port 8080 (dual-stack IPv4/IPv6)');
});
